modules/custom/openy_focal_point/src/Form/OpenYFocalPointCropForm.php

```diff
@@ -0,0 +1,218 @@
+      $url = $this->buildUrl($style, $file, $focal_point_value);
+
+      $derivative_images[$style->id()] = [
+        'style' => $style_label,
+        'url' => $url,
+        'image' => [
+          '#theme' => 'image',
+          '#uri' => $url,
+          '#alt' => $this->t('OpenY Focal Point Preview: %label', ['%label' => $style_label]),
+          '#attributes' => [
+            'class' => ['focal-point-derivative-preview-image'],
+          ],
+        ],
+      ];
+
+      $form['openy_focal_point_preview'] = [
+        '#theme' => "openy_focal_point_preview",
+        '#data' => [
+          'derivative_images' => $derivative_images,
+        ]
+      ];
+
+      // We assume that the last effect is a manual crop.
+      $effects = $style->getEffects()->getConfiguration();
+      $manual = array_pop($effects);
+      $crop_type = $manual['data']['crop_type'];
+
+      $form_state->set('crop_type', $crop_type);
+
+      $form[$style->id()] = [
+        '#type' => 'image_crop',
+        '#file' => $file,
+        '#crop_type_list' => [$crop_type],
+        '#crop_preview_image_style' => 'crop_thumbnail',
+        '#show_default_crop' => FALSE,
+        '#show_crop_area' => TRUE,
+        '#warn_multiple_usages' => FALSE,
+        '#crop_types_required' => [$crop_type],
+      ];
+    }
+
+    $form['submit'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Save Crop'),
+      '#ajax' => [
+        'callback' => '::ajaxSave',
+      ],
+    ];
+
+    $form['remove'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Remove cropping (use default focal point)'),
+      '#ajax' => [
+        'callback' => '::ajaxRemoveCropping',
+      ],
+    ];
+
+    $form['close_dialog'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Close'),
+      '#ajax' => [
+        'callback' => '::closePopup',
+      ],
+    ];
+
+    $form['#attached']['library'][] = 'openy_focal_point/openy_focal_point';
+
+    return $form;
+  }
+
+  public static function ajaxSave(array $form, FormStateInterface $form_state) {
+    /** @var \Drupal\file\Entity\File $file */
+    $file = $form_state->get('file');
+
+    $crop_type = $form_state->get('crop_type');
+
+    $input = $form_state->getUserInput();
+    $crop_properties = $input['prgf_teaser']['crop_wrapper'][$crop_type]['crop_container']['values'];
+    $x = (int) ($crop_properties['x'] + $crop_properties['width'] / 2);
+    $y = (int) ($crop_properties['y'] + $crop_properties['height'] / 2);
+
+    $crop = Crop::findCrop($file->getFileUri(), $crop_type);
+    if ($crop) {
+      if ($crop_properties['height'] == 0 && $crop_properties['width'] == 0) {
+        $crop->delete();
+        $crop = NULL;
+      }
+      else {
+        $crop->setSize($crop_properties['width'], $crop_properties['height']);
+        $crop->setPosition($x, $y);
+      }
+    }
+    else {
+      $crop_storage = \Drupal::entityTypeManager()->getStorage('crop');
```

Comment:
we should pass service via DI

Reply:
@podarok we use this service in public static method, so we don't have instance of the form. This is why we can not use DI here.